internal/proxy/proxy.go

```diff
@@ -510,19 +510,29 @@ func NewClient(ctx context.Context, d cloudsql.Dialer, l cloudsql.Logger, conf *
 	for _, inst := range conf.Instances {
 		m, err := c.newSocketMount(ctx, conf, pc, inst)
 		if err != nil {
-			for _, m := range mnts {
-				mErr := m.Close()
-				if mErr != nil {
-					l.Errorf("failed to close mount: %v", mErr)
+			switch err.(type) {
```

Comment:
Instead of switching on error type, could we just continue if the socket type is a Unix socket (like what we had before)?

Reply:
It will miss the case when the type is TCP socket and there is no active instance. Correct me if I'm wrong, my understanding from the discussion offline:
- Inactive instance with unix socket should fail
- Inactive instance with TCP socket with no port number configured should fail.